addons/web/static/src/views/fields/state_selection/state_selection_field.js

```diff
@@ -94,11 +94,16 @@ export const stateSelectionField = {
             type: "boolean",
             viewTypes: ["list"],
         },
+        {
+            label: _t("Show label"),
+            name: "show_label",
+            type: "boolean",
+        },
     ],
     supportedTypes: ["selection"],
     extractProps({ options, viewType }, dynamicInfo) {
         return {
-            showLabel: viewType === "list" && !options.hide_label,
+            showLabel: (viewType === "list" && !options.hide_label) || !!options.show_label,
```

Comment:
This option currently only affects list views. Should we change that and all existing views that currently rely on it to be hidden on non list views?

Reply:
💯 agree with @brboi indeed, having 2 options that do the opposite thing is just plainly weird

however i would work in the opposite way to keep the diff minimal:
- make the 'hide_label' option true by default (since this is how all views currently treat this widget)
- convert list views to show the label when needed

so basically
`showLabel: !options.hide_label`
+ adding `options="{'hide_label': False}"` in list views.

That way we add the possibility to control the label in any view type, not just tree/form.

I agree that the syntax is gonna be a bit annoying (because you'll need to set 'hide_label' to false to show the label, which is a bit counterintuitive), but since this has been the default everywhere but in list, it seems simpler to convert it in that direction. In addition, it's only used in 6 list views AFAICT, so it's easier to modify that way.

You can then also pbly stop registering it separately for list view in the fields registry?

![image](https://github.com/odoo/odoo/assets/9530767/60eda9fd-1b98-45e1-b40c-ef51c7458000)

Reply:
I agree, if the impact of changing the default in that way doesn't have that big of an impact, this is definitely the cleaner way to go.

Reply:
@bouvyd @robcoekaerts   Currently we are doing in master version. Should we do that in 17.0?

Reply:
I'd say if it doesn't require an upgrade script, do it in 17.0

Reply:
@robcoekaerts @brboi 
If default is true then we have to hide by options. state_selection widget is used a lot of places.like kanban
`<field name="kanban_state" widget="state_selection" options="{'hide_label': True}"/>`

 
what do you think ?

Reply:
yes indeed, let's go with 17